### PR TITLE
Refactor spring-web library instrumentation to Instrumenter API

### PR DESCRIPTION
**Pull request by mateuszrzeszutek**
part of https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/2713

### CODE REVIEW

File: instrumentation/spring/spring-web-3.1/library/README.md [hunk 1]
```diff
@@ -51,16 +55,17 @@ implementation("org.springframework:spring-web:SPRING_VERSION")
...
 ### Features
 
-#### RestTemplateInterceptor
+#### Telemetry-producing `ClientHttpRequestInterceptor` implementation
 
-RestTemplateInterceptor adds OpenTelemetry client spans to requests sent using RestTemplate by implementing the [ClientHttpRequestInterceptor](https://docs.spring.io/spring/docs/current/javadoc-api/org/springframework/http/client/ClientHttpRequestInterceptor.html)
-interface. An example is shown below:
+`SpringWebTracing` allows creating a
+custom [ClientHttpRequestInterceptor](https://docs.spring.io/spring/docs/current/javadoc-api/org/springframework/http/client/ClientHttpRequestInterceptor.html)
+that produces telemetry for HTTP requests sens using a `RestTemplate`. Example:
```
Comment (trask) [inline instrumentation/spring/spring-web-3.1/library/README.md:62]: sens?

**Review by trask — APPROVED** [review]

File: instrumentation/spring/spring-web-3.1/library/README.md (hunk 1 above)
Comment (mateuszrzeszutek) [reply instrumentation/spring/spring-web-3.1/library/README.md:62]: oops, typo

**Comment by trask** [comment]
@mateuszrzeszutek not sure what's going on with CI, you may have to send fresh PR? 😭

**Comment by mateuszrzeszutek** [comment]
> @mateuszrzeszutek not sure what's going on with CI, you may have to send fresh PR? 😭

Sure, will do 👍

**Pull request closed without merging**